File: DI/DI-Dagger-Java/src/main/java/model/persona/controllers/PersonaController.java
package model.persona.controllers;

import model.persona.model.Persona;
import model.persona.repositories.IPersonaRepository;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.UUID;

@Singleton
public class PersonaController {
    private final IPersonaRepository personaRepository;
    private final String id = UUID.randomUUID().toString();

    @Inject
    public PersonaController(IPersonaRepository personaRepository) {
        this.personaRepository = personaRepository;
    }

    public Persona save(Persona persona) {
        System.out.println("PersonaController.save() --> " + persona);
        return personaRepository.save(persona);
    }

    @Override
    public String toString() {
        return "PersonaController{" +
                "id='" + id + '\'' +
                ", personaRepository=" + personaRepository +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/personas/controllers/PersonaController.kt
package models.personas.controllers

import models.personas.model.Persona
import models.personas.repositories.IPersonaRepository
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton


@Singleton
class PersonaController
@Inject constructor(
    private val personaRepository: IPersonaRepository
) {
    private val id = UUID.randomUUID().toString()

    fun save(persona: Persona): Persona {
        println("PersonaController.save() --> $persona")
        return personaRepository.save(persona)
    }

    override fun toString() = "PersonaController(personaRepository=$personaRepository, id='$id')"
}
